### PR TITLE
Check permanent session state in backend as well

### PR DESCRIPTION
**Pull request by hynix**
Fixed permanent session submit params injection from environment vars

### CODE REVIEW

File: server/src/main/java/com/exacaster/lighter/application/sessions/SessionHandler.java [hunk 1]
```diff
@@ -67,7 +69,9 @@ public void keepPermanentSession() {
             return;
         }
         var session = sessionService.fetchOne(sessionId);
-        if (session.isEmpty() || session.filter(it -> it.getState().isComplete()).isPresent()) {
+        var running = not(ApplicationState::isComplete);
+        if (session.map(Application::getState).filter(running).isEmpty() ||
+                backend.getInfo(session.get()).map(ApplicationInfo::getState).filter(running).isEmpty()) {
```
Comment (pdambrauskas) [inline server/src/main/java/com/exacaster/lighter/application/sessions/SessionHandler.java:74]: `session.get()` can return null. Maybe we should check `session.isPresent()` too?

Comment (pdambrauskas) [reply server/src/main/java/com/exacaster/lighter/application/sessions/SessionHandler.java:74]: or even:
`session.map(backend::getInfo).map(ApplicationInfo::getState)....`

Comment (hynix) [reply server/src/main/java/com/exacaster/lighter/application/sessions/SessionHandler.java:74]: > `session.get()` can return null. Maybe we should check `session.isPresent()` too?

session.get() can't be null here - checked before OR.

Comment (pdambrauskas) [reply server/src/main/java/com/exacaster/lighter/application/sessions/SessionHandler.java]: before OR you checked if `running` is empty, not original session.

Comment (hynix) [reply server/src/main/java/com/exacaster/lighter/application/sessions/SessionHandler.java]: > before OR you checked if `running` is empty, not original session.

both. if session is empty before filter `running` - it remains empty anyway :)

**Review by Minutis — APPROVED** [review]
LGTM

File: server/src/main/java/com/exacaster/lighter/application/sessions/SessionHandler.java (hunk 1 above)
Comment (pdambrauskas) [reply server/src/main/java/com/exacaster/lighter/application/sessions/SessionHandler.java]: oh, right 🤦

**Pull request merged**